index.js

```diff
@@ -29,7 +26,7 @@ http.createServer(app).listen(80);
 
 //global salt that is added to every ip before hashing to
 //  make it even harder for someone to decode the ip
-var globalSalt = "49cb0d52-1aec-4b89-85fc-fab2c53062fb";
+var globalSalt = "49cb0d52-1aec-4b89-85fc-fab2c53062fb"; // Should not be global
```

Comment:
If you can figure out how to do ip rate limiting without a global salt, this can be changed. However, this should cause no issues since the DB is private, so the hashing is only a preventative measure in case of a server hack.

FYI: the globalSalt used on the server is different from the one in the code here.

Reply:
Problem is you already have data hashed with the globalSalt it would have to be migrated
And come please switch to bcrypt then you dont even need a salt column

Reply:
Nah, the current rate limiting is a daily thing so I have been deleting the whole IP table every little while.

Reply:
IP rate limiting is security through ignorance even without the privacy problems

Reply:
lol best comment ever.

> IP rate limiting is security through ignorance even without the privacy problems